cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -109,20 +111,83 @@
 
 
   protected String tablePrefix;
+  protected NamespaceQueryAdmin namespaceQueryAdmin;
 
   public void setCConf(CConfiguration cConf) {
     if (cConf != null) {
       this.tablePrefix = cConf.get(Constants.Dataset.TABLE_PREFIX);
     }
   }
 
+  public void setNamespaceQueryAdmin(NamespaceQueryAdmin namespaceQueryAdmin) {
+    if (namespaceQueryAdmin != null) {
+      this.namespaceQueryAdmin = namespaceQueryAdmin;
+    }
+  }
+
+  public Map<String, String> getHBaseToCDAPNamespaceMap() throws IOException {
```

Comment:
although this method name is fairly explanatory, can you add a javadoc specifying why this mapping is needed?